src/python/pants/backend/native/subsystems/native_compile_settings.py

```diff
@@ -0,0 +1,54 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.subsystem.subsystem import Subsystem
+
+
+class NativeCompileSettings(Subsystem):
+  """Any settings relevant to a compiler invocation."""
+
+  default_header_file_extensions = None
+  default_source_file_extensions = None
+
+  @classmethod
+  def register_options(cls, register):
+    super(NativeCompileSettings, cls).register_options(register)
+
+    # TODO: have some more formal method of mirroring options between a target and a subsystem?
+    register('--strict-deps', type=bool, default=True, fingerprint=True, advanced=True,
+             help='The default for the "strict_deps" argument for targets of this language.')
+    register('--fatal-warnings', type=bool, default=True, fingerprint=True, advanced=True,
+             help='The default for the "fatal_warnings" argument for targets of this language.')
+
+    # TODO: make a list of file extension option type?
+    register('--header-file-extensions', type=list, default=cls.default_header_file_extensions,
```

Comment:
It feels a bit premature to make the "file extensions" into options. Certainly it would take subclassing to define a new compiler, in which case having them in options would be redundant.

Reply:
I can understand the idea here, but the compile task will error out if some source files don't have the right extensions -- this is so that we can differentiate headers and sources. Not making them options means users have a restricted set of file names allowed for header and source files unless they make their own subclass in a plugin (and porting a project to Pants might require some work). The alternative is to have separate `headers=` and `sources=` fields, which is very doable -- I just thought this was a cleaner way to obtain headers and sources, and not restrict file names, and removes some (unnecessary?) repetitive typing for the writer of the BUILD file compared to the alternative of splitting `headers=` and `sources=` into separate fields.